proxy/ControlBase.cc

```diff
@@ -796,58 +795,120 @@ ControlBase::ProcessModifiers(matcher_line *line_info)
   _mods.clear();
   _mods.reserve(n_elts);
 
-  // As elements are consumed, the labels are nulled out and the element
-  // count decremented. So we have to scan the entire array to be sure of
-  // finding all the elements. We'll track the element count so we can
-  // escape if we've found all of the elements.
-  for (int i = 0; n_elts && ME_UNKNOWN == err && i < MATCHER_MAX_TOKENS; ++i) {
-    Modifier *mod = nullptr;
-
-    char *label = line_info->line[0][i];
-    char *value = line_info->line[1][i];
-
-    if (!label) {
-      continue; // Already use.
+  if (!line_info->config_is_yaml) {
```

Comment:
@SolidWallOfCode  I wanted to add YAML support making it optional, you only need to change the filename used from parent.config to parent.yaml to use YAML.  It doesn't change the behavior of the current parent.config parser which is per line, that remains.  For YAML, the entire file is loaded and you have a YAML  sequence of maps basically for each  parent entry.  I have the ControlMatcher just reading enough of the YAML, each YAML Node in the sequence, so that it can allocate space by destination type, just as is currently done for the line by line config file.  Once space has been allocated for each destination type, HostMatcher, IPMatcher, URLMatcher etc... will make a call to ParentSelection::Init() to parse the parent selection specific stuff which is then added to the appropriate matchers data structure.

ControlMatcher is parsing line-by-line now.  With this PR, it will be able to use the YAML format as well.

Reply:
I made changes here and removed the unnecessary config_is_yaml bool.  I am passing a pointer to the root yaml node though on each matcher_line but, it's ether all yaml or all old style line format.  The two are not mixed.